packages/compiler/src/process.ts

```diff
@@ -122,15 +122,37 @@ const compileMap = async (def: any, opts: any): Promise<IMapType> => {
   }
 }
 
-const compileFilter = async (def: any): Promise<IFilterType> => ({
+const compileFilterValue = (def: any): IFilterValueNullType | IFilterValueStringType | IFilterValueNumberType | IFilterValueBoolType => {
+  if (def === null) return { type: "mesg.types.Value_NullValue", value: {} }
+  if (typeof def === 'number') return { type: "mesg.types.Value_NumberValue", value: { number_value: def } }
+  if (typeof def === 'boolean') return { type: 'mesg.types.Value_BoolValue', value: { bool_value: def } }
+  if (typeof def === 'string') return { type: 'mesg.types.Value_StringValue', value: { string_value: def } }
+  throw new Error('unsupported filter value')
```

Comment:
predicate EQ is compatible with any type, included map and list.

Reply:
I added this in the list of todo for this feature https://github.com/mesg-foundation/js-sdk/issues/200. This can be done in another PR